src/components/Asset/Edit/_validation.ts

```diff
@@ -2,6 +2,51 @@ import { FileInfo } from '@oceanprotocol/lib'
 import * as Yup from 'yup'
 import web3 from 'web3'
 import { testLinks } from '../../../@utils/yup'
+import { AlgorithmConsumerParameter } from '@components/Publish/_types'
+import { SchemaLike } from 'yup/lib/types'
+import { paramTypes } from '@components/@shared/FormInput/InputElement/ConsumerParameters'
+
+const validationConsumerParameters: {
+  [key in keyof AlgorithmConsumerParameter]: SchemaLike
+} = {
+  name: Yup.string()
+    .test('unique', 'Parameter names must be unique', (name, context) => {
+      // TODO: revert any
+      // from is not yet correctly typed: https://github.com/jquense/yup/issues/398#issuecomment-916693907
+      const [parentFormObj, nextParentFormObj] = (context as any).from
+      if (
+        !nextParentFormObj?.value?.consumerParameters ||
+        nextParentFormObj.value.consumerParameters.length === 1
+      )
+        return true
+
+      const { consumerParameters } = nextParentFormObj.value
+      const occasions = consumerParameters.filter(
+        (params) => params.name === name
+      )
+      return occasions.length === 1
+    })
+    .required('Required'),
+  type: Yup.string().oneOf(paramTypes).required('Required'),
+  description: Yup.string().required('Required'),
+  label: Yup.string().required('Required'),
+  required: Yup.string().oneOf(['optional', 'required']).required('Required'),
+  default: Yup.mixed().when('required', {
+    is: 'optional',
+    then: Yup.mixed()
+      .nullable()
+      .transform((value) => value || null),
+    otherwise: Yup.mixed().required('Required')
+  }),
+  options: Yup.array().when('type', {
+    is: 'select',
+    then: Yup.array()
+      .of(Yup.object())
+      .min(1, 'At least one option needs to be defined')
```

Comment:
thought: could argue that this need to be at least `2` options, since it's not really a select otherwise :D